### PR TITLE
Connect home to paginated sightings

### PR DESCRIPTION
**Pull request by SuwannaTruss**
https://trello.com/c/xINVCI3k/66-connect-home-page-to-paginated-get-all-sightings-endpoint

Ready for review

### CODE REVIEW

File: WhaleSpotting/Controllers/SightingsController.cs [hunk 1]
```diff
@@ -34,9 +34,9 @@ public async Task<List<SightingResponseModel>> GetAllSightings([FromQuery] PageF
             return await _sightings.GetSightings(pageFilter);
         }
 
-        [Authorize]
-        [HttpGet("/search")]
-        public async Task<ActionResult<List<SightingResponseModel>>> SearchSighting([FromQuery] SearchSightingRequestModel searchSighting, PageFilter pageFilter)
+        //[Authorize]
```
Comment (oskarwilliams) [inline WhaleSpotting/Controllers/SightingsController.cs]: Commented code, pls delete

File: WhaleSpotting/Models/RequestModels/SearchSightingRequestModel.cs [hunk 2]
```diff
@@ -14,18 +14,19 @@ public class SearchSightingRequestModel
 
         [Range(-90.00, 90.00, ErrorMessage = "Latitude must be between -90 and 90 degrees")]
         public double? Latitude { get; set; }
-
         public string Location { get; set; } = null;
 
+
```
Comment (oskarwilliams) [inline WhaleSpotting/Models/RequestModels/SearchSightingRequestModel.cs]: Extra new line?

File: WhaleSpotting/Services/SightingsService.cs [hunk 3]
```diff
@@ -53,6 +53,7 @@ public async Task<List<SightingResponseModel>> SearchSighting(SearchSightingRequ
                 .Where(s => searchSighting.SightedTo == null || s.SightedAt <= searchSighting.SightedTo)
                 .Where(s => searchSighting.OrcaPod == null || s.OrcaPod == searchSighting.OrcaPod)
```
Comment (oskarwilliams) [inline WhaleSpotting/Services/SightingsService.cs]: Just realise, could you make all the string search fields 
```suggestion
                .Where(s => string.IsNullOrEmpty(searchSighting.OrcaPod) || s.OrcaPod == searchSighting.OrcaPod)
```

File: WhaleSpotting/react-app/src/api/apiClient.ts [hunk 4]
```diff
@@ -45,3 +46,21 @@ async function getGetSettings(): Promise<any> {
         headers: !token ? {} : { "Authorization": `Bearer ${token}` }
     };
 }
+
+export async function getConfirmedSightings(search: SearchSightingRequestModel, pageNumber = 1, pageSize = 10): Promise<SightingApiModel[]> {
+    //search = {Key: value, Key: Value}
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/api/apiClient.ts]: Commented code

**Review by oskarwilliams — CHANGES_REQUESTED** [review]

File: WhaleSpotting/react-app/src/api/apiClient.ts [hunk 5]
```diff
@@ -56,3 +57,20 @@ async function getGetSettings(): Promise<any> {
...
+        ${search.latitude ? "&latitude=" + search.latitude : ""}
+        ${search.location ? "&location=" + search.location : ""}
+        ${search.sightedFrom ? "&sightedFrom=" + search.sightedFrom : ""}
+        ${search.sightedTo ? "&sightedTo=" + search.sightedTo : ""}
+        ${search.orcaType ? "&orcaType=" + search.orcaType : ""}
+        ${search.orcaPod ? "&orcaPod=" + search.orcaPod : ""}
+        ${search.confirmed ? "&confirmed=" + search.confirmed : ""}
+        ${pageNumber ? "&pageNumber=" + pageNumber : ""}
+        ${pageSize ? "&pageSize=" + pageSize : ""}
+    `, await getGetSettings())
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/api/apiClient.ts]:
```suggestion
        ${pageSize ? "&pageSize=" + pageSize : ""}`,
    await getGetSettings())
```
Can we move the end `, to the line above

**Review by oskarwilliams — CHANGES_REQUESTED** [review]

File: WhaleSpotting/Models/RequestModels/SearchSightingRequestModel.cs (hunk 2 above)
Comment (SuwannaTruss) [reply WhaleSpotting/Models/RequestModels/SearchSightingRequestModel.cs]: done

File: WhaleSpotting/Services/SightingsService.cs (hunk 3 above)
Comment (SuwannaTruss) [reply WhaleSpotting/Services/SightingsService.cs]: done

File: WhaleSpotting/react-app/src/api/apiClient.ts (hunk 5 above)
Comment (SuwannaTruss) [reply WhaleSpotting/react-app/src/api/apiClient.ts]: done

File: WhaleSpotting/Models/RequestModels/SearchSightingRequestModel.cs (hunk 2 above)
Comment (oskarwilliams) [reply WhaleSpotting/Models/RequestModels/SearchSightingRequestModel.cs]: Nope its not

File: WhaleSpotting/Services/SightingsService.cs [hunk 6]
```diff
@@ -49,8 +49,9 @@ public async Task<List<SightingResponseModel>> SearchSighting(SearchSightingRequ
                 .Where(s => searchSighting.Species == null || s.Species == searchSighting.Species)
                 .Where(s => searchSighting.SightedFrom == null || s.SightedAt >= searchSighting.SightedFrom)
                 .Where(s => searchSighting.SightedTo == null || s.SightedAt <= searchSighting.SightedTo)
-                .Where(s => searchSighting.OrcaPod == null || s.OrcaPod == searchSighting.OrcaPod)
+                .Where(s => string.IsNullOrEmpty(searchSighting.OrcaPod) || s.OrcaPod == searchSighting.OrcaPod)
                 .Where(s => searchSighting.Location == null || s.Location == searchSighting.Location)
```
Comment (oskarwilliams) [inline WhaleSpotting/Services/SightingsService.cs]: This is also a string search field so can you fix this as well

File: WhaleSpotting/react-app/src/api/apiClient.ts [hunk 7]
```diff
@@ -56,3 +57,20 @@ async function getGetSettings(): Promise<any> {
...
+        ${search.latitude ? "&latitude=" + search.latitude : ""}
+        ${search.location ? "&location=" + search.location : ""}
+        ${search.sightedFrom ? "&sightedFrom=" + search.sightedFrom : ""}
+        ${search.sightedTo ? "&sightedTo=" + search.sightedTo : ""}
+        ${search.orcaType ? "&orcaType=" + search.orcaType : ""}
+        ${search.orcaPod ? "&orcaPod=" + search.orcaPod : ""}
+        ${search.confirmed ? "&confirmed=" + search.confirmed : ""}
+        ${pageNumber ? "&pageNumber=" + pageNumber : ""}
+        ${pageSize ? "&pageSize=" + pageSize : ""}`,
+    await getGetSettings())
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/api/apiClient.ts:74]: This should be indented a bit sorry!
```suggestion
        await getGetSettings())
```

**Review by oskarwilliams — CHANGES_REQUESTED** [review]

File: WhaleSpotting/react-app/src/api/apiClient.ts (hunk 7 above)
Comment (SuwannaTruss) [reply WhaleSpotting/react-app/src/api/apiClient.ts:74]: When we move it back to indent 8, it fails lint. So, we keep it to indent 4.

**Review by oskarwilliams — APPROVED** [review]

**Pull request merged**